### ISSUE TITLE
Remove Android BIP sound on start and stop recognizing

### ISSUE THREAD

**Issue by MacKentoch** [wenkesj/react-native-voice#48]
In Android SpeechRecognizer emits (horrible) BIP sound when starting and ending recognition.

This by design but it would be awesome to be able to remove / mute these bips.

**Comment by xavicolomer** [comment]
Any news on that @MacKentoch ? 

I would also like to remove that sound.

**Comment by Noitidart** [comment]
This would be a nice option.

It would also be a nice option to have the bip sound in iOS.

**Comment by ohtangza** [comment]
@ghsdh3409 Ping. Time to sync with upstream?

**Comment by alinematich** [comment]
have u been able to do that?

**Comment by dalvallana** [comment]
I (big) while ago, I made a plugin for Cordova, and came up with a dirty hack to turn the volume off while recognizing and turn it back on when done. It's not the best solution but it works fine for now. I don't know exactly how to implement it now because it's been ages since I work with android, but just for the heck of it, here's the code of that class: https://github.com/dalvallana/ContinuousSpeechRecognizer/blob/master/src/android/ContinuousSpeechRecognizer.java. Look for the `mStreamVolume` variable.

**Comment by roysG** [comment]
any update?

**Comment by Saltan77** [comment]
I also need such a function

**Comment by coder4affine** [comment]
any update?

**Comment by ReJaimes** [comment]
Any update on this??

**Comment by roysG** [comment]
any Updateee?

**Comment by subhendukundu** [comment]
Any update?

**Comment by ducpt2** [comment]
+1

**Comment by vaibhgupta09** [comment]
any update

**Comment by lfoliveir4** [comment]
@MacKentoch Did you solved your problem?

**Comment by vaibhgupta09** [comment]
Hello Guyz 
i found a Hack to do this 
just simply set your phone volume to 0 before listening start.
https://www.npmjs.com/package/react-native-system-setting
check this out

**Comment by MacKentoch** [comment]
@lfoliveir4 no but try @vaibhgupta09 hack.

This is by design in Android: recording audio triggers a BIP sound -system does it - (*should be related to legal reasons I guess*).

 So hacking is the only way to bypass as far as I know

**Comment by GoshaEgorian** [comment]
I'm not sure it's good idea to avoid that sound. This makes user to know about listening and as far as this library can start listen without user action it's better to keep BIP sound turned on.

**Comment by zombie6888** [comment]
But it would be good if I could change that sound. I've seen solution for Android API. Maybe I will try it later

**Comment by Jonjoe** [comment]
bump

**Comment by phongbksneep** [comment]
wait this option

**Comment by Hirurgo** [comment]
any update?

**Comment by abhineetsharmathegreat** [comment]
Any updates?

**Comment by GoshaEgorian** [comment]
Down vote
I really don't want this option exist in any RN project. It's not intuitively way to interact with client.

**Comment by erevos-13** [comment]
you can use the `AUDIO_SERVICE` to manipulate the sound on the mobile on android so when the microphone is on silent the phone and when is done to return to the volume levels that the user has.
here a sample that i have try and work.

The patch encompasses the implementation of an Android-based speech recognition feature in a React Native application. It allows the application to mute background sound or music while the Speech Recognizer is capturing user speech. This has been achieved by employing the AudioManager, a core Android service that handles audio management, including volume and mode.

At the start of the VoiceModule class, a private field for AudioManager and an integer field for StreamVolume is declared. The AudioManager is later used to manage audio properties when the microphone is active.

```
private AudioManager mAudioManager = null;
private int mStreamVolume = 0;
```

The application stores the current music stream volume in the mStreamVolume variable when the VoiceModule object is created. This original volume level is maintained for restoration once the speech recognition operation is completed.
```
AudioManager mAudioManager = (AudioManager) this.reactContext.getSystemService(Context.AUDIO_SERVICE);
mStreamVolume = mAudioManager.getStreamVolume(AudioManager.STREAM_MUSIC);
```

When speech recognition begins, the application mutes the music stream volume. This ensures that the recognition process is not affected by any background music that might be playing.

```
muteStreamVolume();
speech.startListening(intent);
```

After the startListening() function is called, the muteStreamVolume() function mutes the audio stream only if the original volume wasn't 0.

```
private void muteStreamVolume() {
    if(mStreamVolume == 0) {
      return;
    }
    AudioManager mAudioManager = (AudioManager) this.reactContext.getSystemService(Context.AUDIO_SERVICE);
    mAudioManager.setStreamVolume(AudioManager.STREAM_MUSIC, 0, 0);
}
```

As soon as the recognition process is completed, the audio stream volume is restored to its original value. This is done in the onResults() method which is called when speech recognition results are ready.

```
setStreamVolumeBack();
```
The setStreamVolumeBack() function restores the original audio stream volume when called.

```
private void setStreamVolumeBack() {
    AudioManager mAudioManager = (AudioManager) this.reactContext.getSystemService(Context.AUDIO_SERVICE);
    mAudioManager.setStreamVolume(AudioManager.STREAM_MUSIC, mStreamVolume, 0);
}
```


In conclusion, this patch significantly improves the user experience by ensuring that a user's voice command is not drowned out by background music during a voice-activated operation. This subtle yet essential modification enables users to interact more seamlessly with voice-activated applications and commands.

**Comment by safaiyeh** [comment]
This project has been deprecated and archived. Please migrate to [`expo-speech-recognition`](https://github.com/jamsch/expo-speech-recognition). Thank you for your contributions! ❤️
